### PR TITLE
29 finalize filters

### PR DESCRIPTION
**Pull request by kev1n80**
# Overall
- Add documentation
- Order props and css styling in alphabetical order 

## CheckboxButtonGroup
- Abstract `CheckboxButtonGroup`
  - Create `CheckboxFilterButtonGroup` which specifically uses the checkbox buttons to add filters
- `CheckboxButtonGroup` now accepts `icons` and `labelPlacement` props for the checkbox

## OriginCheckboxInput
- `OriginCheckboxInput` now looks similar to the mockups
  - However, the change in styling for the border when the checkbox is checked/unchecked takes a while

## StatusCheckboxInput
- `StatusCheckboxInput` now bolds the label of the checkbox that was checked

## CohortMultiSelectInput
- Changed `CohortMultiSelectInput` from using MaterialUI `Select` components to `Autocomplete` components. User can type inside the filter in order to filter the options. Plus options will appear below the filter component or above it if there is not enough space below the filter component.

## Overall Filters Added
- Display the filters that have been added on the top of the `StudentDrawerFilter` component to match mockup design
  - Use ReactAdmin `useListController` to get information on the number of filters that have been added

# Next Step 
- Use data provider from the services folder to get information on the number of data that matches a filter. This will allow us to mimic the mock up where there is a number next the label.
- Find component to add to GPA range in order to show the number of students that follow between certain increments of GPA to match mockup.
-  Move to Typescript

# IMAGE OF CHANGES
image

### Drop down
image

### Input matches an option
NOTE: The label of the Cohort filter is not removed, it is covered by the drop down menu
image

### Input does not match any of the options
NOTE: The label of the Cohort filter is not removed, it is covered by the drop down menu
image

### Filters on top
image

### CODE REVIEW

**Review by sophiacrennan — APPROVED** [review]
Looks good! Remember to fix the merge conflict in StudentDrawer.

**Comment by kev1n80** [comment]
> Looks good! Remember to fix the merge conflict in StudentDrawer.

Sorry, I just saw this. Yes, I will fix these before I merge this branch B)

File: pharmd-app/src/screens/student/StudentDrawer.js [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
-import { useSelector } from "react-redux";
 import { Route, MemoryRouter } from "react-router";
 import { Link as RouterLink } from "react-router-dom";
+import { useListController } from "react-admin";
+import { useSelector } from "react-redux";
 
 // Component Imports
 import DrawerMaterial from "@material-ui/core/Drawer";
-import tw, { styled } from "twin.macro";
+import StudentQuickView from "./StudentQuickView";
```
Comment (sophiacrennan) [inline pharmd-app/src/screens/student/StudentDrawer.js:20]: This import is already on line 27. Application fails to compile unless one is removed

**Review by sophiacrennan — APPROVED** [review]
Wow awesome job! This looks really good!

File: pharmd-app/src/screens/student/StudentDrawer.js (hunk 1 above)
Comment (kev1n80) [reply pharmd-app/src/screens/student/StudentDrawer.js]: Good catch!

**Review by Jose-S — APPROVED** [review]
Finally looks like the mockups. Thank you :D

**Pull request merged**